src/replication.c

```diff
@@ -1171,6 +1171,21 @@ int replicationAppend(struct raft *r,
      * that we issue below actually completes.  */
     for (j = 0; j < n; j++) {
         struct raft_entry *entry = &args->entries[i + j];
+
```

Comment:
Reasoning out loud here:

The leader has tried to replicate a log entry say with data `D1` at term `T1` and index `I1`, replicates to a majority of followers doesn't hear back from any of them, disk write fails, and now we hit this logic and the leader tries to replicate a new log entry with data `D2` and term `T1` and `I1`, disk write succeeds and leader crashes. I think we now have a situation where there are different log entries for the same term and index.